### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by Osushi** [omkarcloud/google-maps-scraper#168] [Ideas]
Thank you for the best experience through this tool.

I would like you to consider addressing the following points.

**1. Memory pressure**
I am generally scraping about 500 keywords at a time, however, the memory grows infinitely with each use.
If you can, please optimize the memory.
![スクリーンショット 2024-05-24 17 30 02](https://github.com/omkarcloud/google-maps-scraper/assets/6112972/d8e6517a-7e41-4e36-a2aa-3de83fea7d7d)
(This is just an example, but I've experienced use of up to 190GB.)

**2. Batch registration of "Search Queries" on the GUI**
When registering multiple queries on the GUI, I need to press the "Add Field" button for that number of times.
Is there a way to register a batch from a file or other source?

Might be similar to this discussion.
https://github.com/omkarcloud/google-maps-scraper/discussions/160

**3. How to update the Pro version**
I downloaded the latest version from the link in the email I received with my purchase, is this correct?

I used to be able to re-download it from the following page, but now the link is gone for some reason
https://github.com/sponsors/omkarcloud?frequency=one-time

If there is a way to update the correct Pro version, please let me know.

**Comment by Chetan11-dev** [comment]
1. Please run the following commands to resolve the memory bug:

python -m pip install bota botasaurus botasaurus_api botasaurus_driver bota botasaurus-proxy-authentication botasaurus_server --upgrade

Kindly let me know if it reoccurs. 

2. Will Consider
3, Yes, Correct, to update it kindly follow the steps given in email again.

**Reply by Osushi** [reply]
Thanks for your very courteous reply.

I will give it a try.

**Reply by Osushi** [reply]
I've confirmed that it's fixed perfectly.

Thank you very much for the great improvement!

**Reply by Chetan11-dev** [reply]
Welcome!

**Comment by Osushi** [comment]
👏
